docs/guides/development/nodejs/nodejs-twitter-bot/index.md

```diff
@@ -0,0 +1,294 @@
+
+Twitter bots are a great way to practice using an API and developing some coding skills along the way. You can usually get the bot going for under 100 lines of code, so it’s especially good for beginners. In this tutorial, I’ll show you how to use Node.js, *a framework for javascript* and a few npms, or *node package managers* to have your bot tweeting in no time.
+
+## Before you Begin:
+
+You should download or install:
+[Visual Studio Code](https://code.visualstudio.com/)
+[GitHub desktop](https://desktop.github.com/)
+
+Be ready to visit:
+[Twitter](https://twitter.com/home)
+[Twitter Developer portal](https://developer.twitter.com/en)
+[GitHub](https://github.com/)
+[Linode](https://www.linode.com)
+
+You can look at the documentation:
+[Node.js](https://nodejs.org/en/)
+[Twit](https://www.npmjs.com/package/twit)
+[node-schedule](https://www.npmjs.com/package/node-schedule)
+[DotEnv](https://www.npmjs.com/package/dotenv)
+</br>
+
+{{< note >}}
+The steps in this guide require root privileges. Be sure to run the steps below as `root` or with the `sudo` prefix. For more information on privileges, see our [Users and Groups](/docs/tools-reference/linux-users-and-groups/) guide.
+{{< /note >}}
+
+## Setting up a Twitter Account
+
+This is where you decide what you want your account to be named and what it will do automatically. This bot will tweet a link to a video game soundtrack once a week (pulled randomly from an array of links) and will respond to a user who asks for a link with a random link from the same array. We’ll call our bot [@SNESSoundtracks.](https://twitter.com/snessoundtracks) Go to Twitter and follow the instructions for making a brand new account for your bot.
+</br>
+
+## Twitter Developer Portal
+
+The developer portal is where you register your new account to be able to use the Twitter API. This is an important step because it unlocks the accounts ability to tweet by using the API with code instead of manually tweeting. You know, the whole point of a bot?
+
+Head to the developer [page.](https://developer.twitter.com/en) Once there you’ll see a link near the profile picture on the top right that says “apply.”
+
+Click that, and it’ll take you to a page where you apply to have your Twitter profile become an active developer profile. Follow the instructions on screen until you have an authenticated developer account. Once that’s done, create the app and fill out the descriptions for it. All you have to say is that you’re doing it as a hobby and it’s just for learning to code or practicing a bot. As long as you’re not giving info to the government or really consuming streaming data for analytical purposes, this is all pretty straight forward. When you’ve finished, the developer portal should look like this.
+
+![The Twitter Developer page](https://user-images.githubusercontent.com/65838556/115882571-e00d3900-a41a-11eb-83bc-2811066299f2.png)
+</br>
+
+## The Keys and the Code and the Node Package Manager
```

Comment:
I split this into three different sections and expanded each:

Create and Clone a GitHub Repository
https://deploy-preview-4492--nostalgic-ptolemy-b01ab8.netlify.app/docs/guides/nodejs-twitter-bot/#create-and-clone-a-github-repository
This new section includes instructions for using GitHub Desktop and the git command line

Create an App in the Twitter Developer Portal
https://deploy-preview-4492--nostalgic-ptolemy-b01ab8.netlify.app/docs/guides/nodejs-twitter-bot/#create-an-app-in-the-twitter-developer-portal
New section includes an ordered-list for the instructions, an outline of the form questions asked by Twitter, a note about projects vs. apps in the Twitter developer portal, and some more explicit instructions for which keys in the dev portal correspond with which lines in the .env file. Also, I added a section that shows how to set Read and Write permissions and how to generate new access tokens on the new app, because my new test apps were read-only to start.

Initialize the Node Project and Install npm Dependencies
https://deploy-preview-4492--nostalgic-ptolemy-b01ab8.netlify.app/docs/guides/nodejs-twitter-bot/#initialize-the-node-project-and-install-npm-dependencies
Reformatted the instructions for this section as an ordered-list, inserted the links to the NPM module docs here, added some explanation for the package.json file and links to other docs, and inserted "before" versions of the "scripts" section in package.json that gets changed